pkg/asset/machines/azure/machines.go

```diff
@@ -110,15 +110,17 @@ func provider(platform *azure.Platform, mpool *azure.MachinePool, osImage string
 		image.Offer = mpool.OSImage.Offer
 		image.SKU = mpool.OSImage.SKU
 		image.Version = mpool.OSImage.Version
-	} else {
+	} else if useImageGallery {
 		// image gallery names cannot have dashes
 		galleryName := strings.Replace(clusterID, "-", "_", -1)
 		id := clusterID
 		if hyperVGen == "V2" {
 			id += "-gen2"
 		}
-		imageID := fmt.Sprintf("/resourceGroups/%s/providers/Microsoft.Compute/galleries/gallery_%s/images/%s/versions/%s", rg, galleryName, id, rhcosVersion)
+		imageID := fmt.Sprintf("/resourceGroups/%s/providers/Microsoft.Compute/galleries/gallery_%s/images/%s/versions/latest", rg, galleryName, id)
 		image.ResourceID = imageID
+	} else {
+		image.ResourceID = fmt.Sprintf("/resourceGroups/%s/providers/Microsoft.Compute/images/%s", rg, clusterID)
```

Comment:
You need to append `"-gen2"` here to the image name if the VM supports `hyperVGen == "V2"`

Reply:
should be fixed in e26a4783415d32420778300ac5322f6953ca2511